warehouse/core/src/main/java/datawave/iterators/filter/ConfigurableAgeOffFilter.java

```diff
@@ -336,46 +336,39 @@ public void init(SortedKeyValueIterator<Key,Value> source, Map<String,String> op
-                                                                .build(new ReloadableCacheBuilder());
-                                            }
-                                        } catch (Throwable t) {
-                                            log.error(t, t);
+        synchronized (ConfigurableAgeOffFilter.class) {
+            if (null == ruleCache) {
+                UPDATE_INTERVAL_MS = getLongProperty(UPDATE_INTERVAL_MS_PROP, DEFAULT_UPDATE_INTERVAL_MS); // 5 ms
+                EXPIRATION_INTERVAL_MS = getLongProperty(EXPIRATION_INTERVAL_MS_PROP, DEFAULT_EXPIRATION_INTERVAL_MS); // 1 hour
+                ruleCache = CacheBuilder.newBuilder().refreshAfterWrite(UPDATE_INTERVAL_MS, TimeUnit.MILLISECONDS)
+                                .expireAfterAccess(EXPIRATION_INTERVAL_MS, TimeUnit.MILLISECONDS).build(new ReloadableCacheBuilder());
+                // this will schedule a check to see if the update or expiration intervals have changed
+                // if so the ruleCache will be rebuilt with these new intervals
+                SimpleTimer.getInstance(1).schedule(
+                                () -> {
+                                    try {
+                                        long interval = getLongProperty(UPDATE_INTERVAL_MS_PROP, DEFAULT_UPDATE_INTERVAL_MS);
+                                        long expiration = getLongProperty(EXPIRATION_INTERVAL_MS_PROP, DEFAULT_EXPIRATION_INTERVAL_MS);
+                                        if (UPDATE_INTERVAL_MS != interval || EXPIRATION_INTERVAL_MS != expiration) {
+                                            log.info("Changing " + UPDATE_INTERVAL_MS_PROP + " to " + interval);
+                                            UPDATE_INTERVAL_MS = interval;
+                                            log.info("Changing " + EXPIRATION_INTERVAL_MS_PROP + " to " + expiration);
+                                            EXPIRATION_INTERVAL_MS = expiration;
+                                            ruleCache = CacheBuilder.newBuilder().refreshAfterWrite(UPDATE_INTERVAL_MS, TimeUnit.MILLISECONDS)
+                                                            .expireAfterAccess(EXPIRATION_INTERVAL_MS, TimeUnit.MILLISECONDS)
+                                                            .build(new ReloadableCacheBuilder());
                                         }
-                                    }, 1000, 10 * 1000);
-                }
+                                    } catch (Throwable t) {
+                                        log.error(t, t);
+                                    }
+                                }, 1000, 10 * 1000);
             }
         }
         
         Path filePath = new Path(filename);
         if (null == fs) {
             synchronized (ConfigurableAgeOffFilter.class) {
-                if (null == fs) {
```

Comment:
Actually, it might make sense to keep both null checks.  

In the original code, if the fs is not null, then access to the fs resource will never be blocked since the synchronize call is skipped. With these changes, only one thread will be able to access the fs resource at a time, blocking all other threads.

Reply:
moved this block into the `synchronized` block above - since this should only be happening once on load.